premium/usage.go

```diff
@@ -161,14 +153,14 @@ type BatchUpdater struct {
 	isClosed       bool
 }
 
-func NewUsageClient(pluginTeam cqapi.PluginTeam, pluginKind cqapi.PluginKind, pluginName cqapi.PluginName, ops ...UsageClientOptions) (*BatchUpdater, error) {
+func NewUsageClient(meta plugin.Meta, ops ...UsageClientOptions) (UsageClient, error) {
```

Comment:
I agree this signature change is an improvement, but it's technically a breaking change, and we'll have to update all plugins using the usage package. Maybe we should do a v5 release for this?

Reply:
`NewUsageClient` is used only in CQ-authored, effectively private plugins.

Reply:
That might be true but we have community plugins that are premium as well. And if they're not using it, they should be

Reply:
I don't think that warrants an upgrade of ~7000 lines in the public cq/cq repo.

Reply:
Can't we just avoid this by keeping the signature the same? It looks like we can adopt `UsageClientOptions` for the additional option we need

Reply:
Oh I guess the `*BatchUpdater` return might still be an issue. Well let's at least make a clear note in the changelog that this is a breaking change to the premium package interface, and explain what plugin authors need to do to handle it.

Reply:
There are ways to avoid this by shifting the burden to the plugin side, which is not favourable.

Reply:
Yeah we don't need to go that far; let's just add an explanation to the changelog

Reply:
I've updated the PR desc. Adding a `BREAKING-CHANGE` there might bump the version, but further edits or suggestions are welcome.